src/components/AddItemForm.js

```diff
@@ -1,28 +1,64 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import fb from '../lib/firebase';
 import '../css/AddItemForm.css';
-import { v4 as uuidv4 } from 'uuid';
 
 const Form = ({token}) => {
     const [itemName, setItemName] = useState("");
     const [timeFrame, setTimeFrame] = useState(7);
     const [lastPurchaseDate, setPurchaseDate] = useState(null);
     const userToken = token || "userToken";
+    const [shoppingListCollection, setShoppingListCollection] = useState([]);
+    const [duplicateError, setDuplicateError] = useState(false);
+
+    useEffect(() => {
+        const db =  fb.firestore()
+        const tokenRef = db.collection(token);
+
+        tokenRef
+            .orderBy("timeFrame", "asc")
+            .get()
+            .then((querySnapshot) => {
+                let fullCollection = [];
+                if(!querySnapshot.empty){
+                    querySnapshot.forEach((doc) => {
+                        let documentData = doc.data();
+                        let nameData = documentData.itemName
+                        if(!nameData){
+                            nameData = nameData.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"").trim().replace(/\s{2,}/g," ");
+        
+                            fullCollection.push(nameData);
+                        }
+                    });
+                    setShoppingListCollection(fullCollection);
+                }
+            }).catch((error) => {
+                console.log("Error getting document:", error);
+            });
+    }, [token, shoppingListCollection]);
 
```

Comment:
My super bad...we need to remove `shoppingListCollection` from this array, because it is overloading the db. When  the component loads it sets an initial state to `[]`, and then in this `useEffect`, it sets `shoppingListCollection` to whatever is in the db, and then it re-renders and sets `shoppingListCollection` to an empty array, etc. etc. etc. We still need to figure out a way to get the new item into the `shoppingListCollection`, though.  Sorry! @jbatara and @dkimlim . I will think on it if you all can, too.

Reply:
I've got an idea that actually works and doesn't overload the db.  : ) Note, this is AN approach, and it seems to work with both an existing list and a brand new list, but you may come up with a better way, for sure.

1. Move the code inside the `useEffect` into a function like this:
`const getCurrentShoppingListItems = (userToken) => {
   const db = fb.firestore()
   // etc. etc. --- all of the code that's inside the useEffect
}`

2.Inside the `useEffect` call this function: `getCurrentShoppingListItems(userToken)` ---this will ensure that we get the db items on page load.
3.pass `userToken` into the `useEffect` dependency array.
4. Inside your `.then` of the `handleSubmit` call `getCurrentShoppingListItems(userToken)` something like this:
`.then((docRef) => { tokenRef.doc(docRef.id).update({ id : docRef.id });  getCurrentShoppingListItems(userToken)})`
It needs to be inside the `.then`. This ensures that the item for sure has been added to the db before we pull back the new list.
5. Test this out locally.  Never assume that your mentor/code reviewer is correct.  : )

Let me know if you want to zoom, @jbatara and @dkimlim